src/sentry/eventstore/models.py

```diff
@@ -246,8 +246,15 @@ def title(self) -> str:
         if column in self._snuba_data:
             return cast(str, self._snuba_data[column])
 
-        et = eventtypes.get(self.get_event_type())()
-        return cast(str, et.get_title(self.get_event_metadata()))
+        title = self.data.get("title")
```

Comment:
is top level title supposed to take precedence over meta-data title?

Reply:
The levels of indirection are enough to drive you mad, but they're the same value.

In `_materialize_metadata_many`, we first get the eventtype, use that to get the event metadata, and then feed to it `materialize_metadata`:

https://github.com/getsentry/sentry/blob/a51f5c70b12e83f11ec917c40b34401bbd1979ce/src/sentry/event_manager.py#L869-L879

In `materialize_metadata`, we feed that metadata to the eventtype's `get_title` method to get the actual `title` value:

https://github.com/getsentry/sentry/blob/a51f5c70b12e83f11ec917c40b34401bbd1979ce/src/sentry/event_manager.py#L1316-L1322

Back in `_materialize_metadata_many`, we then add that into `data` (which is the event data):

https://github.com/getsentry/sentry/blob/a51f5c70b12e83f11ec917c40b34401bbd1979ce/src/sentry/event_manager.py#L879

So what we've been doing in the `title` property is just recreating that process exactly. All I'm doing is making it so we only go through it once, rather than at least twice.